diff -uNr dhcp-4.2.0.orig//common/bpf.c dhcp-4.2.0/common/bpf.c
--- dhcp-4.2.0.orig//common/bpf.c	2010-08-24 21:05:43.000000000 +0200
+++ dhcp-4.2.0/common/bpf.c	2010-08-24 21:08:59.000000000 +0200
@@ -358,9 +358,9 @@
 	struct iovec iov [3];
 	int result;
 
-	if (!strcmp (interface -> name, "fallback"))
+/*	if (!strcmp (interface -> name, "fallback"))
 		return send_fallback (interface, packet, raw,
-				      len, from, to, hto);
+				      len, from, to, hto); */
 
 	if (hto == NULL && interface->anycast_mac_addr.hlen)
 		hto = &interface->anycast_mac_addr;
@@ -537,7 +537,7 @@
 
 void maybe_setup_fallback ()
 {
-	isc_result_t status;
+/*	isc_result_t status;
 	struct interface_info *fbi = (struct interface_info *)0;
 	if (setup_fallback (&fbi, MDL)) {
 		if_register_fallback (fbi);
@@ -548,7 +548,7 @@
 			log_fatal ("Can't register I/O handle for %s: %s",
 				   fbi -> name, isc_result_totext (status));
 		interface_dereference (&fbi, MDL);
-	}
+	} */
 }
 
 void
diff -uNr dhcp-4.2.0.orig//includes/osdep.h dhcp-4.2.0/includes/osdep.h
--- dhcp-4.2.0.orig//includes/osdep.h	2010-08-24 21:05:43.000000000 +0200
+++ dhcp-4.2.0/includes/osdep.h	2010-08-24 21:06:07.000000000 +0200
@@ -157,7 +157,7 @@
    Currently, all low-level packet interfaces use BSD sockets as a
    fallback. */
 
-#if defined (USE_BPF_SEND) || defined (USE_NIT_SEND) || \
+#if defined (USE_NIT_SEND) || \
     defined (USE_DLPI_SEND) || defined (USE_UPF_SEND) || \
     defined (USE_LPF_SEND) || \
     (defined (USE_SOCKET_SEND) && defined (HAVE_SO_BINDTODEVICE))
